code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/ServiceDeploymentPlanner.java

```diff
@@ -28,7 +28,7 @@ public ServiceDeploymentPlanner(List<Service> services, List<DeploymentUnit> uni
 
         if (units != null) {
             for (DeploymentUnit unit : units) {
-                if (unit.isUnhealthy()) {
+                if (unit.isUnhealthy() || !unit.isHostActive()) {
```

Comment:
I can make this change.  My original thought was DeploymentUnit should just supply the info and DeploymentPlanner can take whatever actions it wants using that info.  But in this function, I guess it's just grouping into healthy and unhealthy anyways.